Mage.Sets/src/mage/cards/h/HinataDawnCrowned.java

```diff
@@ -0,0 +1,122 @@
+        this.power = new MageInt(4);
+        this.toughness = new MageInt(4);
+
+        // Flying
+        this.addAbility(FlyingAbility.getInstance());
+
+        // Trample
+        this.addAbility(TrampleAbility.getInstance());
+
+        // Spells you cast cost {1} less to cast for each target.
+        this.addAbility(new SimpleStaticAbility(new HinataDawnCrownedOwnEffect()));
+
+        // Spells your opponents cast cost {1} more to cast for each target
+        this.addAbility(new SimpleStaticAbility(new HinataDawnCrownedOpponentsEffect()));
+    }
+
+    private HinataDawnCrowned(final HinataDawnCrowned card) {
+        super(card);
+    }
+
+    @Override
+    public HinataDawnCrowned copy() {
+        return new HinataDawnCrowned(this);
+    }
+}
+
+class HinataDawnCrownedOwnEffect extends CostModificationEffectImpl {
+
+    HinataDawnCrownedOwnEffect() {
+        super(Duration.WhileOnBattlefield, Outcome.Neutral, CostModificationType.REDUCE_COST);
+        staticText = "Spells you cast cost {1} less to cast for each target";
+    }
+
+    private HinataDawnCrownedOwnEffect(HinataDawnCrownedOwnEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public HinataDawnCrownedOwnEffect copy() {
+        return new HinataDawnCrownedOwnEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source, Ability abilityToModify) {
+        CardUtil.reduceCost(abilityToModify, HinataDawnCrownedEffectUtility.getTargetCount(game, abilityToModify));
+        return true;
+    }
+
+    @Override
+    public boolean applies(Ability abilityToModify, Ability source, Game game) {
+        return abilityToModify instanceof SpellAbility
+                && abilityToModify.isControlledBy(source.getControllerId());
+    }
+}
+
+class HinataDawnCrownedOpponentsEffect extends CostModificationEffectImpl {
+
+    HinataDawnCrownedOpponentsEffect() {
+        super(Duration.WhileOnBattlefield, Outcome.Neutral, CostModificationType.INCREASE_COST);
+        staticText = "Spells your opponents cast cost {1} more to cast for each target";
+    }
+
+    private HinataDawnCrownedOpponentsEffect(HinataDawnCrownedOpponentsEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public HinataDawnCrownedOpponentsEffect copy() {
+        return new HinataDawnCrownedOpponentsEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source, Ability abilityToModify) {
+        CardUtil.increaseCost(abilityToModify, HinataDawnCrownedEffectUtility.getTargetCount(game, abilityToModify));
+        return true;
+    }
+
+    @Override
+    public boolean applies(Ability abilityToModify, Ability source, Game game) {
+        return abilityToModify instanceof SpellAbility
+                && !abilityToModify.isControlledBy(source.getControllerId());
+    }
+}
+
+final class HinataDawnCrownedEffectUtility
+{
+    public static int getTargetCount(Game game, Ability abilityToModify)
```

Comment:
this should probably be a static method of one of the other classes rather than its own class but I don't think it matters

Reply:
You are right, I don't think it matters. Its just my stylistic choice. I can move it into one of the effect classes if you prefer, but if you don't care, Ill just leave it as is.